File: src/main/java/com/example/demo/Service/UserService.java
package com.example.demo.Service;

import com.example.demo.DTO.LoginDTO;
import com.example.demo.DTO.SignUpDTO;
import com.example.demo.Model.User;
import com.example.demo.Repository.UserRepo;
import com.example.demo.Response.CustomResponse;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.util.Date;
import java.util.HashSet;
import java.util.Optional;
import java.util.Set;

@Service
public class UserService{
    @Autowired
    private UserRepo userRepository;

    public ResponseEntity<?> registerUser(SignUpDTO signUpDTO,boolean isAdmin) {
        Optional<User> existedUser = userRepository.findByEmail(signUpDTO.getEmail());
        if (existedUser.isPresent()){
            return ResponseEntity.status(HttpStatus.CONFLICT).body(new CustomResponse(HttpStatus.CONFLICT.value(), "User already exists", null));
        }
        User newUser = new User();
        newUser.setUsername(signUpDTO.getUsername());
        newUser.setEmail(signUpDTO.getEmail());
        newUser.setPassword(signUpDTO.getPassword());
        newUser.setCreated(new Date());
        String userRole = "user";
        newUser.setRole(userRole);
        userRepository.save(newUser);
        return ResponseEntity.status(HttpStatus.CREATED).body(new CustomResponse(HttpStatus.CREATED.value(), "created",null));
    }

    public ResponseEntity<?> userLogin(LoginDTO loginDTO){
        Optional<User> existedUser = userRepository.findByEmail(loginDTO.getEmail());
        if (existedUser.isEmpty()){
            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(new CustomResponse(HttpStatus.NOT_FOUND.value(), "User not found", null));
        }
        if (!loginDTO.getEmail().equals(existedUser.get().getEmail())){
            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(new CustomResponse(HttpStatus.NOT_FOUND.value(), "user does not existed with email",null));

        } else if (!loginDTO.getPassword().equals(existedUser.get().getPassword())) {
            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(new CustomResponse(HttpStatus.BAD_REQUEST.value(), "incorrect password! try again",null));

        }
        return ResponseEntity.status(HttpStatus.OK).body(new CustomResponse(HttpStatus.OK.value(), "login success",existedUser));

    }


}

File: bank/src/main/kotlin/com/junocoding/bank/service/userService.kt
package com.junocoding.bank.service
import com.junocoding.bank.DTOS.userRegister
import com.junocoding.bank.models.User
import com.junocoding.bank.repositories.userRepository
import org.springframework.stereotype.Service
import org.springframework.web.bind.annotation.RequestBody
import java.util.*
import kotlin.NoSuchElementException

@Service
class userService (private val userRepository: userRepository){

    fun save(user :User):User{
        return userRepository.save(user)
    }
    fun findByEmail(email:String):User?{
        return this.userRepository.findByEmail(email)

    }
    fun findUser(id:Long): Optional<User> {
        return userRepository.findById(id)
    }
    fun updateUser(id:Long,@RequestBody body:userRegister): User {
        val existeduser=userRepository.findById(id).orElseThrow { NoSuchElementException("No user found")}
        body.name.let { existeduser.name=it }
        body.email.let { existeduser.email=it }
        body.password.let { existeduser.password=it }
        val updatedUser=userRepository.save(existeduser)
        return updatedUser
    }
}